### PR TITLE
CrunchyDB Test Suite and required Keywords added and updated.

### PR DESCRIPTION
**Pull request by harshkumarRH**
With this PR, the following have been covered - 

- First Test Suite for Crunchy Bridge (currently covers-Creation of Provider Account with valid and invalid credentials, deploying DBaas Instance from the created Provider Account)
- Moves the content of Page Object Resource file _resources/object_repo/database_access_obj.resource_ to _resources/object_repo/create_dbaasinventory.resource_
- Parameterization of Keywords where possible
- Addition of Invalid Credentials Test Case to MongoDB Test Suite

Marks tentative completion of - [DBaaS-521](https://issues.redhat.com/browse/DBAAS-521) and [DBaaS-523](https://issues.redhat.com/browse/DBAAS-523)

Signed-off-by: Harsh Kumar <hakumar@redhat.com>

### CODE REVIEW

File: resources/keywords/create_provider_account.resource [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+
 *** Variables ***
 ${provaccname}      ${EMPTY}
+${INV_ORG_ID}       invalid_org_id
+${INV_PUB_KEY}      invalid_pub_key
+${INV_PRI_KEY}      invalid_private_key
 
 *** Keywords ***
-User Filters Project Other Than redhat-dbaas-operator And Navigates To Database Access Page
+User Filters Project Other Than Default DBaas Operator And Navigates To Database Access Page
```
Comment (mrrajan) [inline resources/keywords/create_provider_account.resource]: Shall we update this as, "User Selects Invalid Namespace For Provider Account Creation And Navigates To Database Access Page"

Screenshot from 2022-05-13 09-47-40

File: resources/keywords/deploy_instance_dev.resource [hunk 2]
```diff
@@ -3,15 +3,16 @@ Resource    ../keywords/create_provider_account.resource
...
+${newProject}        ${EMPTY}
 ${instanceName}      ${EMPTY}
 
 *** Keywords ***
 User Selects A Project For Database Connection
     Run Keyword If    '${newProject}'=='${EMPTY}'
     ...    Create New Project
-    ...  ELSE
+    ...  ELSE    Run Keywords
     ...    Log To Console    Project ${newProject} exists
```
Comment (mrrajan) [inline resources/keywords/deploy_instance_dev.resource:14]: My bad, we can remove the "Log To Console" keyword

File: tests/crunchydb_create_connect.robot [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Suite Setup         Set Library Search Order    SeleniumLibrary
+Suite Teardown      Tear Down The Test Suite
+Test Setup          Given The Browser Is On Openshift Home Screen
+Test Teardown       Close Browser
+
+
+*** Test Cases ***
+Scenario: Verify error message for invalid credentials on CrunchyDB
+    [Tags]    smoke    RHOD-49-2
```
Comment (mrrajan) [inline tests/crunchydb_create_connect.robot:16]: Can you please confirm on the JIRA ID for on line 16

File: tests/mongodb_create_connect.robot [hunk 4]
```diff
@@ -12,20 +12,29 @@ Test Teardown       Close Browser
 
 *** Test Cases ***
 Scenario: Error Message To Select Valid Namespace For Provider Account Creation
-    [Tags]    smoke
-    When User Filters Project Other Than redhat-dbaas-operator And Navigates To Database Access Page
+    [Tags]    smoke    RHOD-45
+    Skip
```
Comment (mrrajan) [inline tests/mongodb_create_connect.robot:16]: Can you please confirm whether we move this test to different robot file or skipping for now?

File: resources/object_repo/installed_operators_obj.resource [hunk 5]
```diff
@@ -6,8 +6,9 @@ Object Repository for Installed Operators Screen
 ${elemDbaasStatus}          //a[contains(.,"OpenShift Database Access Operator")]//ancestor::tr/td[contains(.,"Succeeded")]
 ${titleDbaasOperator}       //h1[contains(.,"OpenShift Database Access Operator") and contains(@class,"logo__name")]
 ${elemDbaasInvStatus}       //td[contains(.,"<<paname>>")]/parent::tr/td[contains(.,"SpecSynced, InventoryReady")]
+${titleDbaasInvertories}    //h1[contains(.,"DBaaSInventories")]
+${lnkProvAcc}               //a[contains(.,"OpenShift Database Access Operator")]//ancestor::tr/td/div/a[contains(.,"Provider Account")]
```
Comment (mrrajan) [inline resources/object_repo/installed_operators_obj.resource]: Shall we update the xpath to - "//a[contains(.,"OpenShift Database Access Operator")]//ancestor::tr//a[contains(.,"Provider Account")]" to cut down the relative dependency.

**Review by mrrajan — CHANGES_REQUESTED** [review]
@harshkumarRH I have reviewed the PR and added inline comments, LMKWYT. And Please share the logs/results of the dry run.

File: tests/crunchydb_create_connect.robot (hunk 3 above)
Comment (harshkumarRH) [reply tests/crunchydb_create_connect.robot:16]: So as per the consolidated Smoke Tests Sheet, we have only one scenario for Verifying error message for Invalid Credentials, and hence the corresponding Stage Jira ID for that test case is RHOD-49
Ideally I should have added two more identical Test Cases for Crunchy and Cockroach on Stage Jira and updated tags accordingly, but I didn't get a chance, so as a temporary placeholder I just marked the tags as RHOD-49-1 for MongoDB and RHOD-49-2 for CrunchyDB for now
These will be updated with subsequent PRs

Comment (mrrajan) [reply tests/crunchydb_create_connect.robot:16]: Thanks!

File: tests/mongodb_create_connect.robot (hunk 4 above)
Comment (harshkumarRH) [reply tests/mongodb_create_connect.robot:16]: Before raising the PR, I planned to save the output log for mongo and crunchy separately and wanted them to be clean, I was hoping to discuss with the team before moving this test case to another Test Suite
So in the dilemma of not moving the Test Case and also wanting the logs to be clean, I just skipped it for the time being.
Please rest assured this is a very temporary change and will be amended with later PRs

File: resources/keywords/deploy_instance_dev.resource (hunk 2 above)
Comment (harshkumarRH) [reply resources/keywords/deploy_instance_dev.resource:14]: Can we keep this around for a little while, just until we cover all the corner cases that we might encounter.
This helps a bit in debugging and we can remove this in subsequent PRs, will that be okay?

File: resources/object_repo/installed_operators_obj.resource (hunk 5 above)
Comment (harshkumarRH) [reply resources/object_repo/installed_operators_obj.resource]: Thank you for the suggestion. Tried, tested and updated with new commit.

File: resources/keywords/create_provider_account.resource (hunk 1 above)
Comment (harshkumarRH) [reply resources/keywords/create_provider_account.resource]: Done, updated

**Comment by harshkumarRH** [comment]
@mrrajan Thank you for your comments, I have addressed them, please review
Also, the logs -  [PR23_Log.zip](https://github.com/RHODA-lab/rhoda-ci/files/8684551/PR23_Log.zip)

File: tests/mongodb_create_connect.robot (hunk 4 above)
Comment (mrrajan) [reply tests/mongodb_create_connect.robot:16]: Thanks for the clarification

File: resources/keywords/deploy_instance_dev.resource (hunk 2 above)
Comment (mrrajan) [reply resources/keywords/deploy_instance_dev.resource:14]: Sure

**Review by mrrajan — APPROVED** [review]
LGTM

File: resources/keywords/create_provider_account.resource [hunk 6]
```diff
@@ -36,55 +41,130 @@ Application Navigate To Create Provider Account Page And Error Message Displayed
...
 
 Application Navigates To Provider Account Creation Success Screen
     Wait Until Keyword Succeeds     5x     2s      Wait Until Page Contains Element    ${lblDBInstSuccess}
     Element Should Be Visible    ${btnViewProviderAcc}
 
-Provider Account Creation Success
+Provider Account Creation Success Page
     Click Element    ${btnViewProviderAcc}
-    Wait Until Page Contains Element    ${btnLastUpdatedHeader}     timeout=10s
+    Wait Until Page Contains Element    ${titleDbaasInvertories}    timeout=10s
```
Comment (ikanias) [inline resources/keywords/create_provider_account.resource]: I think there is a spelling error here:  ${titleDbaasInvertories}. I think it should be  ${titleDbaasInventories}?

File: resources/keywords/create_provider_account.resource [hunk 7]
```diff
@@ -36,55 +41,130 @@ Application Navigate To Create Provider Account Page And Error Message Displayed
...
+    Check Status Of Invalid Provider Account
+
+Provider Account Creation Failure Screen
+    Wait Until Page Contains Element    ${titleCreateProviderAcc}    timeout=15s
+    Wait Until Page Contains Element     ${lblDBInstFailure}    timeout=20s
+    Page Should Contain Button    ${btnEditProvAcc}
+
+User Navigates To DBaaS Inventories Screen
+    User Navigates To Installed Operators Under Operators
+    User Filters redhat-dbaas-operator Namespace On Project Dropdown
```
Comment (ikanias) [inline resources/keywords/create_provider_account.resource]: Shouldn't we use ${PROJECT} like before?

**Review by ikanias — APPROVED** [review]
LGTM just added a few comments for clariffication.

File: resources/keywords/create_provider_account.resource (hunk 7 above)
Comment (harshkumarRH) [reply resources/keywords/create_provider_account.resource]: Parameterized Project Namespace using variable _${PROJECT}_ with default value _redhat-dbaas-operator_

File: resources/keywords/create_provider_account.resource (hunk 6 above)
Comment (harshkumarRH) [reply resources/keywords/create_provider_account.resource]: Thanks @ikanias, I have changed it to ${titleDbaasInventories} everywhere.
Pushed with latest commit

**Review by mrrajan — APPROVED** [review]

**Review by ikanias — APPROVED** [review]
LGTM, approved.

**Pull request merged**